open-craft/frontend-app-course-authoring
# [FC-0059] Components tab

## Description

This PR adds:
- Library components card
- Card Grid in Library Home - Components tab.
- Connect with API to get library components.

## Supporting information

- Github Issue: https://github.com/openedx/frontend-app-course-authoring/issues/1037
- Internal ticket: [FAL-3757](https://tasks.opencraft.com/browse/FAL-3757).

## Testing instructions

- Enable [library authoring](https://github.com/openedx/openedx-tutor-plugins/tree/main/plugins/tutor-contrib-library-authoring-mfe) in your tutor.
- Use this branch of edx-platform and follow the testing instructions to setup: https://github.com/openedx/edx-platform/pull/35000
- Go to the [library authoring](http://apps.local.edly.io:3001/library-authoring/) and create an new library
- In library authoring create components, more that 40 to test Infinite scroll. TODO Fixtures to create test content?
- Edit a title and description of a component to be long.
- Run tutor dev run cms bash and `./manage.py cms reindex_studio --experimental`.
- Go to the library home using the id of your new library: http://apps.local.edly.io:2001/course-authoring/library/{library-id}
- Go to componentas tab
- Check that the components cards are loaded.
- Check that the cards match with designs. Check that long title and log description are truncated.
- Scroll down and check the infinite scroll.
